dbt/adapters/spark_cde/cdeapisession.py

```diff
@@ -205,14 +223,48 @@ def execute(self, sql: str, *parameters: Any) -> None:
         self._rows = rows
         self._schema = schema
 
-        # 7. cleanup resources
+        # 7.fetch spark events
+        logger.debug("{}: Get spark events".format(job_name))
+        self.get_spark_job_events(job_name, job)
+        logger.debug("{}: Done get spark events".format(job_name))
+
+        # 8. cleanup resources
         logger.debug("{}: Delete job".format(job_name))
         self._cde_connection.delete_job(job_name)
         logger.debug("{}: Done delete job".format(job_name))
         logger.debug("{}: Delete resource".format(job_name))
         self._cde_connection.delete_resource(job_name)
         logger.debug("{}: Done delete resource".format(job_name))
 
+    """
+    Fetch spark events from log and log them along with their corresponding timestamps
+    """
+
+    def get_spark_job_events(self, job_name, job):
```

Comment:
There is no return for the method?

Reply:
No. Should i implicily return none. 

This just prints the spark events. Something similar to this.

`08:32:56.182517 [debug] [ThreadPool]: Spark adapter: dbt-job-1662107436674-00000465: Get spark events
08:32:56.182804 [debug] [ThreadPool]: Spark adapter: dbt-job-1662107436674-00000465: Get spark job events
08:32:56.183133 [debug] [ThreadPool]: Spark adapter: dbt-job-1662107436674-00000465: Sleep for 40 seconds
08:33:36.151693 [debug] [ThreadPool]: Spark adapter: dbt-job-1662107436663-00000546: Done sleep for 40 seconds
08:33:36.187750 [debug] [ThreadPool]: Spark adapter: dbt-job-1662107436674-00000465: Done sleep for 40 seconds
08:33:37.720605 [debug] [ThreadPool]: Spark adapter: dbt-job-1662107436674-00000465: Done get spark job events
08:33:37.721959 [debug] [ThreadPool]: Spark adapter: dbt-job-1662107436674-00000465: SparkListenerBlockManagerAdded           08:31:25.858000                         
08:33:37.722261 [debug] [ThreadPool]: Spark adapter: dbt-job-1662107436674-00000465: SparkListenerApplicationStart            08:31:18.249000                         
08:33:37.722532 [debug] [ThreadPool]: Spark adapter: dbt-job-1662107436674-00000465: org.apache.spark.sql.execution.ui.SparkListenerSQLExecutionStart 08:31:53.546000                         
08:33:37.722797 [debug] [ThreadPool]: Spark adapter: dbt-job-1662107436674-00000465: org.apache.spark.sql.execution.ui.SparkListenerSQLExecutionEnd 08:31:54.007000                         
08:33:37.723329 [debug] [ThreadPool]: Spark adapter: dbt-job-1662107436674-00000465: org.apache.spark.sql.execution.ui.SparkListenerSQLExecutionStart 08:31:54.121000                         
08:33:37.723703 [debug] [ThreadPool]: Spark adapter: dbt-job-1662107436674-00000465: org.apache.spark.sql.execution.ui.SparkListenerSQLExecutionEnd 08:31:54.152000                         
08:33:37.725010 [debug] [ThreadPool]: Spark adapter: dbt-job-1662107436674-00000465: SparkListenerApplicationEnd              08:31:54.184000                         
08:33:37.725376 [debug] [ThreadPool]: Spark adapter: dbt-job-1662107436674-00000465: Done get spark events`

Reply:
Ya, got that. Probably you can simply change the method name to log_spark_job_events as the rest of the get* methods return something